main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala

```diff
@@ -333,7 +333,35 @@ object SemanticTokensProvider {
 
     case Expr.Use(_, _, exp, _) => visitExp(exp) // TODO NS-REFACTOR add token for sym
 
-    case Expr.Cst(_, _, _) => Iterator.empty
+    case Expr.Cst(Constant.Str(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.String, Nil, loc))
+
+    case Expr.Cst(Constant.Char(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.String, Nil, loc))
+
+    case Expr.Cst(Constant.Int8(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(Constant.Int16(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(Constant.Int32(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(Constant.Int64(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(Constant.BigInt(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(Constant.Float32(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(Constant.Float64(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(Constant.BigDecimal(_), _, loc) => Iterator(SemanticToken(SemanticTokenType.Number, Nil, loc))
+
+    case Expr.Cst(_: Constant.Regex, _, loc) => Iterator(SemanticToken(SemanticTokenType.Regexp, Nil, loc))
+
+    case Expr.Cst(_: Constant.Bool, _, loc) => Iterator(SemanticToken(SemanticTokenType.Type, Nil, loc))
+
+    case Expr.Cst(Constant.Unit, _, loc) => Iterator(SemanticToken(SemanticTokenType.Type, Nil, loc))
+
+    case Expr.Cst(Constant.Null, _, loc) => Iterator(SemanticToken(SemanticTokenType.Type, Nil, loc))
+
+    case Expr.Cst(Constant.RecordEmpty, _, loc) => Iterator(SemanticToken(SemanticTokenType.Type, Nil, loc))
```

Comment:
Also please keep the pattern match in the exact same order as the constants are listed in `Constant.scala` -- it helps maintainability.